google/genai/_api_client.py

```diff
@@ -506,11 +515,74 @@ async def async_request_streamed(
     response = await self._async_request(http_request=http_request, stream=True)
 
     async def async_generator():
-      async for chunk in response:
-        yield chunk
+      async for line in response.aiter_lines():
+        if line:
+          yield json.loads(line)
 
     return async_generator()
 
+  def download_file(self, path: str, http_options):
+    """Downloads the file data.
+
+    Args:
+      path: The request path with query params.
+      http_options: The http options to use for the request.
+
+    returns:
+          The file bytes
+    """
+    http_request = self._build_request(
+        'get', path=path, request_dict={}, http_options=http_options
+    )
+    return self._download_file_request(http_request).byte_stream[0]
+
+  def _download_file_request(
+      self,
+      http_request: HttpRequest,
+  ) -> HttpResponse:
+    data = None
+    if http_request.data:
+      if not isinstance(http_request.data, bytes):
+        data = json.dumps(http_request.data, cls=RequestJsonEncoder)
+      else:
+        data = http_request.data
+
+    http_session = requests.Session()
+    response = http_session.request(
```

Comment:
If it's a single use only, you don't need to create the session, because it creates already internally. If you are going to reuse the session, then it makes sense.
```suggestion
    response = requests.request(
```